src/test/tools/bzlmod/MODULE.bazel.lock

```diff
@@ -82,13 +82,14 @@
             }
           }
         },
+        "moduleExtensionMetadata": null,
```

Comment:
Not sure how we feel about `null` values popping up here.

Reply:
Can you explain a bit why `null` ended up here?

Reply:
That's because of 
https://github.com/bazelbuild/bazel/pull/22498/files#diff-11c1c9cda0a3bafa6f377878111faecc4be3f83ac0a74fe9115f3cb297d78bf2R251

I could add a dedicated type adapter for `Optional<String>` to prevent this from happening for other `Optional<T>`. What do you think?

Reply:
Yeah, I think that would be a cleaner solution?

Reply:
Done!